velox/common/memory/MappedMemory.cpp

```diff
@@ -313,59 +315,45 @@ int64_t MappedMemoryImpl::freeNonContiguous(Allocation& allocation) {
     return 0;
   }
   MachinePageCount numFreed = 0;
-  if (FLAGS_velox_use_malloc) {
-    for (int32_t i = 0; i < allocation.numRuns(); ++i) {
-      PageRun run = allocation.runAt(i);
-      numFreed += run.numPages();
-      void* ptr = run.data();
-      {
-        std::lock_guard<std::mutex> l(mallocsMutex_);
-        if (mallocs_.find(ptr) == mallocs_.end()) {
-          VELOX_CHECK(false, "Bad free");
-        }
-        mallocs_.erase(ptr);
-      }
-      stats_.recordFree(
-          std::min<int64_t>(
-              sizeClassSizes_.back() * kPageSize, run.numPages() * kPageSize),
-          [&]() {
-            ::free(ptr); // NOLINT
-          });
+  for (int32_t i = 0; i < allocation.numRuns(); ++i) {
+    PageRun run = allocation.runAt(i);
+    numFreed += run.numPages();
+    void* ptr = run.data();
+    {
+      std::lock_guard<std::mutex> l(mallocsMutex_);
+      const auto ret = mallocs_.erase(ptr);
+      VELOX_CHECK_EQ(ret, 1, "Bad free page pointer: ", ptr);
+      mallocs_.erase(ptr);
     }
-  } else {
-    throw std::runtime_error("Not implemented");
+    stats_.recordFree(
+        std::min<int64_t>(
+            sizeClassSizes_.back() * kPageSize, run.numPages() * kPageSize),
+        [&]() {
+          ::free(ptr); // NOLINT
+        });
   }
   numAllocated_.fetch_sub(numFreed);
   allocation.clear();
   return numFreed * kPageSize;
 }
 
 void MappedMemoryImpl::freeContiguousImpl(ContiguousAllocation& allocation) {
-  if (allocation.data() && allocation.size()) {
-    if (munmap(allocation.data(), allocation.size()) < 0) {
-      LOG(ERROR) << "munmap returned " << errno << "for " << allocation.data()
-                 << ", " << allocation.size();
-    }
-    numMapped_.fetch_sub(allocation.numPages());
-    numAllocated_.fetch_sub(allocation.numPages());
-    allocation.reset(nullptr, nullptr, 0);
+  if (allocation.data() == nullptr || allocation.size() == 0) {
+    return;
+  }
+  if (::munmap(allocation.data(), allocation.size()) < 0) {
+    LOG(ERROR) << "munmap returned " << errno << "for " << allocation.data()
+               << ", " << allocation.size();
   }
+  numMapped_.fetch_sub(allocation.numPages());
+  numAllocated_.fetch_sub(allocation.numPages());
+  allocation.reset(nullptr, nullptr, 0);
 }
 
 bool MappedMemoryImpl::checkConsistency() const {
-  if (FLAGS_velox_use_malloc) {
```

Comment:
Hi Meng,

  Right now, we rely on this flag since we want to use malloc for MappedMemory and avoid the dependency on mmap (on account of different OS use). What will be the alternative if this code is removed ?

Reply:
This flag doesn't take effective for now. PrestoCPP always use mmap? We can check on this in PrestoCPP instead?